src/inmanta/compiler/__init__.py

```diff
@@ -120,6 +120,17 @@ def anchormap(refs: Dict[Any, Any] = {}) -> Sequence[Tuple[Location, Location]]:
     return sched.anchormap(compiler, statements, blocks)
 
 
+def get_types_and_scopes() -> Tuple[Dict[str, inmanta_type.Type], Namespace]:
+    """
+        Only run the compilation steps required to extract the different types and scopes.
+    """
+    compiler = Compiler()
+    (statements, blocks) = compiler.compile()
+    sched = scheduler.Scheduler(compiler_config.track_dataflow())
+    sched.define_types(compiler, statements, blocks)
```

Comment:
I think we should catch any `CompilerException` here and call `compiler.handle_exception()`. It writes errors to the configured file.

Reply:
Would it not be more logical for the caller of this method to use handle_exception when this is appropriate? 

I think most other pieces of software that use this method are happier with and exception than with a file?

Reply:
Indeed, that makes sense. It does seem somewhat inconsistent that we do it differently for `do_compile` but I can see why we would do it.

Reply:
good remark, perhaps something to add as a note to the compiler SDK epic that we should have methods for consumption in programs and methods for consumption in CLI context

Reply:
I've added it to the epic description